src/test/java/org/eclipse/uprotocol/communication/SubscriberExample.java

```diff
@@ -0,0 +1,39 @@
+package org.eclipse.uprotocol.communication;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import org.junit.jupiter.api.Test;
+import org.eclipse.uprotocol.transport.UListener;
+import org.eclipse.uprotocol.transport.UTransport;
+import org.eclipse.uprotocol.v1.UMessage;
+import org.eclipse.uprotocol.v1.UUri;
+
+public class SubscriberExample {
+    @Test
+    public void testSubscribe() {
+        // Topic to subscribe to
+        UUri topic = UUri.newBuilder()
+            .setUeId(4).setUeVersionMajor(1).setResourceId(0x8000)
+            .build();
+
+        // Listener to receive published messages on
+        final UListener myListener = new UListener() {
+            @Override
+            public void onReceive(UMessage message) {
+                // Handle receiving subscriptions here
+            }
+        };
+
+        // The target transport
+        UTransport transport = new TestUTransport();
+        
+        // Subscribe
+        assertFalse(Subscriber.subscribe(transport, topic, myListener)
+            .toCompletableFuture().isCompletedExceptionally());
+        
+        // Unsubscribe
+        Subscriber.unsubscribe(transport, topic);
```

Comment:
subscribe adds the listener, why does unsubscribe not remove the listener and we have to do it manually.
I find this a place for leaky listeners lying around even when a client is unsubscribed.
How am I supposed to remove the listener if I did not add it

Reply:
I need the unregisterListener() separate so we can be permanently subscribed without the listener but I see your point about also having an API to remote the listener at the same time.  I'll add this.